File: Defense_of_a_Kingdom.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main(){
    int t;
    cin >> t;
    while(t--){
        int xdim, ydim, n;
        cin >> xdim >> ydim >> n;
        if(n==0){
            cout << xdim*ydim << endl;
            continue;
        }
        vector<int> X(n+2);
        vector<int> Y(n+2);
        for (; n>0; n--){
            int x, y;
            cin >> x >> y;
            X.push_back(x);
            Y.push_back(y);
        }
        X.push_back(0);
        X.push_back(xdim+1);
        Y.push_back(0);
        Y.push_back(ydim+1);

        sort(X.begin(), X.end());
        int x_win = 0;
        for (int i = 0; i < X.size()-1; i++){
            x_win = max(x_win, X[i+1]-X[i]-1);
        }
        sort(Y.begin(), Y.end());
        int y_win = 0;
        for (int i = 0; i < Y.size()-1; i++){
            y_win = max(y_win, Y[i+1]-Y[i]-1);
        }

        cout << y_win*x_win << endl;
    }
    return 0;
}

File: Defense_of_a_Kingdom.py
import itertools as it

t = int(input())
for _ in range(t):
    x_dim, y_dim, n = input().split()
    x_dim = int(x_dim)
    y_dim = int(y_dim)
    n = int(n)
    if n == 0:
        print(x_dim*y_dim)
        continue
    towers = [tuple(map(lambda x: int(x), input().split())) for _ in range(n)]
    towers.append((0,0))
    towers.append((x_dim+1, y_dim+1))
    
    towers.sort(key = lambda x: x[0])
    x_win = max(map(
        lambda s: s[1]-s[0]-1, 
        it.pairwise((x[0] for x in towers))
        ))

    towers.sort(key = lambda x: x[1])
    y_win = max(map(
        lambda s: s[1]-s[0]-1, 
        it.pairwise((x[1] for x in towers))
        ))
    
    print(y_win*x_win)
    